src/main/java/dev/sigstore/KeylessSigner.java

```diff
@@ -0,0 +1,185 @@
+            FulcioVerifier.newFulcioVerifier(fulcioCert, ctfePublicKey))
+        .rekorClient(RekorClient.builder().build(), RekorVerifier.newRekorVerifier(rekorPublicKey))
+        .oidcClient(WebOidcClient.builder().build())
+        .signer(Signers.newEcdsaSigner());
+    return builder;
+  }
+
+  public static Builder builderForStaging()
+      throws IOException, InvalidAlgorithmParameterException, CertificateException,
+          InvalidKeySpecException, NoSuchAlgorithmException {
+    var builder = new Builder();
+    var fulcioCert =
+        Resources.toByteArray(Resources.getResource("dev/sigstore/tuf/staging/fulcio.crt.pem"));
+    var ctfePublicKey =
+        Resources.toByteArray(Resources.getResource("dev/sigstore/tuf/staging/ctfe.pub"));
+    var rekorPublicKey =
+        Resources.toByteArray(Resources.getResource("dev/sigstore/tuf/staging/rekor.pub"));
+    builder
+        .fulcioClient(
+            FulcioClient.builder()
+                .setServerUrl(URI.create(FulcioClient.STAGING_FULCIO_SERVER))
+                .build(),
+            FulcioVerifier.newFulcioVerifier(fulcioCert, ctfePublicKey))
+        .rekorClient(
+            RekorClient.builder()
+                .setServerUrl(URI.create(RekorClient.STAGING_REKOR_SERVER))
+                .build(),
+            RekorVerifier.newRekorVerifier(rekorPublicKey))
+        .oidcClient(WebOidcClient.builder().setIssuer(WebOidcClient.STAGING_DEX_ISSUER).build())
+        .signer(Signers.newEcdsaSigner());
+    return builder;
+  }
+
+  public static Builder builder() {
+    return new Builder();
+  }
+
+  public static class Builder {
+    private FulcioClient fulcioClient;
+    private FulcioVerifier fulcioVerifier;
+    private RekorClient rekorClient;
+    private RekorVerifier rekorVerifier;
+    private OidcClient oidcClient;
+    private Signer signer;
+
+    public Builder fulcioClient(FulcioClient fulcioClient, FulcioVerifier fulcioVerifier) {
+      this.fulcioClient = fulcioClient;
+      this.fulcioVerifier = fulcioVerifier;
+      return this;
+    }
+
+    public Builder rekorClient(RekorClient rekorClient, RekorVerifier rekorVerifier) {
+      this.rekorClient = rekorClient;
+      this.rekorVerifier = rekorVerifier;
+      return this;
+    }
+
+    public Builder oidcClient(OidcClient oidcClient) {
+      this.oidcClient = oidcClient;
+      return this;
+    }
+
+    public Builder signer(Signer signer) {
+      this.signer = signer;
+      return this;
+    }
+
+    public KeylessSigner build() {
+      Preconditions.checkNotNull(fulcioClient);
+      Preconditions.checkNotNull(fulcioVerifier);
+      Preconditions.checkNotNull(rekorClient);
+      Preconditions.checkNotNull(rekorVerifier);
+      Preconditions.checkNotNull(oidcClient);
+      Preconditions.checkNotNull(signer);
+      return new KeylessSigner(
+          fulcioClient, fulcioVerifier, rekorClient, rekorVerifier, oidcClient, signer);
+    }
+  }
+
+  public void sign(Path artifact)
```

Comment:
should this return the rekor response?

Reply:
Should probably return a signature, a public key and a rekor response.